src/components/EventDetails/EventDetails.js

```diff
@@ -1328,9 +1337,13 @@ export class EventDetail extends React.Component<Props> {
           }
         }
     }
+
+    if (!eventData) return null;
```

Comment:
Please check that EventDetails works correctly if null is returned here?

Reply:
It does, it just doesn't render the bottom modal. The thing is it's actually set to `null` before `switch` and it would still return null if it didn't change after `switch` which won't happen as `default` still adds some data over it. Just to avoid more noise with additional checks in the code below where `eventData` has final appends and I just went straight forward with this check. You're aware that the code here is very messy, but didn't want to rewrite it (plus the fact that you changed history components with V2). Hope it makes sense.